oneapp-server/types/ApplicationFoodStampInfo.js

```diff
@@ -42,7 +42,7 @@ const typeDef = gql`
 
```

Comment:
Did you check to see if any of the `ApplicationFoodStampInfoInput` fields have validation rules from the legacy ASP.net that need to be enforced?

We have the [graphql-constraint-directive](https://www.npmjs.com/package/graphql-constraint-directive) available, so we can mark them in the type definitions and write additional constraints if need be.

The one I'm thinking about most is fields that depend on values of other fields. I'm not sure if any fields in this type need a validation rule like that, but I know we have them in this project.

Thoughts on how we should handle them? Not sure if there are pre-built constraints for fields like that, or we should deviate our type model to handle that situation better. For example, remove the `HAS_RECEIVED_EMERGENCY_FS` field and replace it with a `EMERGENCY_FS` input type that has the `EMERGENCY_FS_*` fields under it. If `EMERGENCY_FS` is omitted, we assume `HAS_RECEIVED_EMERGENCY_FS` is false, but if it's included, we can rely on GraphQL type validation or required fields by marking the input type fields as required. Thoughts?

Reply:
Reviewed the [existing model](https://github.com/newjersey/dhs-oneapp-legacy/blob/080a57380e9dedd834d2b6de141d7a031825159e/NJOneAppAuto/Forms/Application/FoodStampInformation.aspx) and added requirements (!) to required fields. 

I agree that a separate `EMERGENCY_FS` type would make sense, but am hesitant to move around the model type too much. My intuition is that for the first MVP, we keep these data types in their original hierarchy, and we can look to clean them up in the next pass?

Reply:
Sounds good to me. I'm wondering if we should than clear the `EMERGENCY_FS_*` fields if `HAS_RECEIVED_EMERGENCY_FS` is false? I'm unsure what the downstream systems would do if `HAS_RECEIVED_EMERGENCY_FS` is false and there is data persisted in the other fields.

Reply:
good call. I've enforced logic in the mutation to clear out the `EMERGENCY_FS` fields if `HAS_RECEIVED_EMERGENCY_FS` is false